workflow/usage.rst

```diff
@@ -222,6 +222,38 @@ could easily enable logging::
     $subscriber = new AuditTrailListener($logger);
     $dispatcher->addSubscriber($subscriber);
 
+Available events
+----------------
+
+1. General events
+
+The following events are dispatched for all workflows:
+
+* ``workflow.guard``: occurs just before starting a transition. It allows you to
+  prevent the transition by calling ``$event->setBlocked(true);`` as shown above.
+* ``workflow.leave``: occurs just after an object has left it's current state.
```

Comment:
We should mention that all events give carries the subject and the transition.
Also I think we should clarify what state is available in the event:
```rst
* ``workflow.leave``: carries the marking with the initial places, occurs just before the transition.
* ``workflow.transition``: carries the marking with the current places, occurs during the transition.
* ``workflow.enter``: carries the marking with the new places, occurs just after the transition.
```
Actually while checking this I've opened https://github.com/symfony/symfony/pull/21793.